src/main/java/com/autotune/common/datasource/DataSourceInfo.java

```diff
@@ -26,10 +26,14 @@
  * }
  */
 public class DataSourceInfo {
+    private final String name;
+    private final String serviceName;
     private final String provider;
     private final URL url;
 
-    public DataSourceInfo(String provider, URL url) {
+    public DataSourceInfo(String name, String serviceName, String provider, URL url) {
```

Comment:
add more parametes like
HTTP Method:
Headers
Authentication Parameters:
Query Parameters:
 based o AUth type you may have to consider some more parameters or class